src/components/scroll-monitor.tsx

```diff
@@ -43,13 +44,24 @@ class ScrollMonitor extends React.Component<Props, null> {
       this.lastChild = null
     }
 
+    if (this.props.refresh != lastProps.refresh) {
+      this.resetScroll()
+      this.check()
+    }
```

Comment:
I don't really love this. I think this code could be improved by passing a controller object of some kind into the scroll monitor. Need to think about this more.

In any case, this fixes the bug in a pretty safe, straightforward way.